File: beepi/py/sort_note_freq_toops.py
#!/usr/bin/python

#####################################
# @author: vladimir kulyukin
#####################################
import sys
import re

note_freq_toops = [ ]
for line in sys.stdin.readlines():
    match = re.match(r'\S+', line)
    if match != None:
        note_freq_toop = re.split(r'\t', line)
        note_freq_toops.append(note_freq_toop)

note_freq_toops.sort(key=lambda toop: int(toop[1]), reverse=True)
for note, freq in note_freq_toops:
    sys.stdout.write(note+"\t"+str(freq))

File: beepi/pl/sort_note_freq_toops.pl
#!/usr/bin/perl

###############################
## @author: vladimir kulyukin
###############################    
use strict;
use warnings;
use sort '_quicksort';
use sort 'stable';

my @note_freq_toops = ();
while ( <> ) {
    if ( $_ =~ /\S+/ ) {
	my @note_freq_toop = split(/\t/, $_);
	push(@note_freq_toops, \@note_freq_toop);
    }
}
print "@{$_}[0]\t@{$_}[1]" foreach(sort { @{$b}[1] <=> @{$a}[1] } @note_freq_toops);
